core/domain/html_validation_service.py

```diff
@@ -831,3 +831,216 @@ def get_filename_with_dimensions(old_filename, exp_id):
     new_filename = regenerate_image_filename_using_dimensions(
         old_filename, height, width)
     return new_filename
+
+
+def add_image_id_in_image_tag(
+        html, image_counter):
+    """Adds image_id-with-value in image tag.
+
+    Args:
+        html: str. HTML of content in which image tag present.
+        image_counter: int. Counter for an image id.
+
+    Returns:
+        (new_content_html, image_counter): Tuple. Tuple contains new html
+            of content and image_id counter.
+    """
+    soup = bs4.BeautifulSoup(html.encode('utf-8'), 'html.parser')
+    for image in soup.findAll(name='oppia-noninteractive-image'):
+        image_counter += 1
+        image['image_id-with-value'] = image_counter
+
+    new_content_html = unicode(soup)
+
+    return (new_content_html, image_counter)
+
+
+def get_image_info_from_html(html):
+    """Extracts info of each image present in the HTML.
+
+    Args:
+        html: str. the HTML in which the images are present.
+
+    Returns:
+        image_info_list: list. List contaning an image info object.
+    """
+    image_info_list = []
+    soup = bs4.BeautifulSoup(html.encode('utf-8'), 'html.parser')
+    for image in soup.findAll(name='oppia-noninteractive-image'):
+        image_info = {
+            'id': image['image_id-with-value'],
+            'info': {
+                'src': image['filepath-with-value'],
+                'placeholder': False,
+                'author_id': '',
+                'instructions': ''
+            }
+        }
+        image_info_list.append(image_info)
+
+    return image_info_list
+
+
+def remove_filepath_from_image_tag(html):
+    """Removes filepath attribute from image tag.
+
+    Args:
+        html: str. The HTML in which the image tag presents.
+
+    Returns:
+        html: str. the HTML in which image tag presents without filepath
+            attribute.
+    """
+    soup = bs4.BeautifulSoup(html.encode('utf-8'), 'html.parser')
+    for image in soup.findAll(name='oppia-noninteractive-image'):
+        del image['filepath-with-value']
+
+    return unicode(soup)
+
+
+def add_image_id_and_image_assets_in_state_dict(state_dict, image_counter):
```

Comment:
As discussed in the meeting try to send `html` data to the html related services.

Reply:
done